drain3/template_miner.py

```diff
@@ -173,21 +173,28 @@ def match(self, log_message: str, full_search_strategy="never") -> LogCluster:
     def get_parameter_list(self, log_template: str, content: str):
         escaped_prefix = re.escape(self.config.mask_prefix)
         escaped_suffix = re.escape(self.config.mask_suffix)
-        template_regex = re.sub(escaped_prefix + r".+?" + escaped_suffix, self.drain.param_str, log_template)
-        if self.drain.param_str not in template_regex:
-            return []
-        template_regex = re.escape(template_regex)
-        template_regex = re.sub(r'\\ +', r'\\s+', template_regex)
-        template_regex = "^" + template_regex.replace(escaped_prefix + r"\*" + escaped_suffix, "(.*?)") + "$"
+        content_masked =  re.sub(escaped_prefix + r".+?" + escaped_suffix, lambda match: self.config.mask_prefix + "*" + self.config.mask_suffix, content)
+        for instruction in self.config.masking_instructions:
+            content_masked = instruction.regex.sub(lambda match: self.config.mask_prefix + "(" + re.escape(match.group(0)) + ")" + self.config.mask_suffix, content_masked)
+        masking_parameter_list = [re.sub(r"\\(.)", r"\1", match.group(1)) for match in re.finditer(escaped_prefix + r"\((.+?)(?<!\\)\)" + escaped_suffix, content_masked)]
```

Comment:
The parameters are extracted (in the correct order!) from all masks containing parameters (which can be easily identified as the only masks like `<(...)>`).

The escaping of the parameter is undone now.